assertj-core/src/main/java/org/assertj/core/util/introspection/CaseStyleInsensitiveFieldLookupStrategy.java

```diff
@@ -0,0 +1,36 @@
+package org.assertj.core.util.introspection;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.recursive.comparison.ComparingSnakeOrCamelCaseFields.COMPARING_SNAKE_OR_CAMEL_CASE_FIELDS;
+import static org.assertj.core.util.Preconditions.checkArgument;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Looks up fields by comparing their name previously converted into a common case style.
+ */
+class CaseStyleInsensitiveFieldLookupStrategy implements FieldLookupStrategy {
```

Comment:
`CaseStyleInsensitive` is too generic here as we only take care of snake/camelCase so it is better to rename that `SnakeOrCamelCaseFieldLookupStrategy` to avoid misleading users.